converter/diag/diagnostic.go

```diff
@@ -0,0 +1,34 @@
+// Package diag exposes error types used throughout converter and a method to
+// pretty-print them to the screen.
+package diag
+
+import (
+	"fmt"
+)
+
+// Diagnostic is an individual diagnostic message. Diagnostic messages can have
+// different levels of severities.
+type Diagnostic struct {
+	// Severity holds the severity level of this Diagnostic.
+	Severity Severity
+
+	Summary string
+	Detail  string
+}
+
+var _ fmt.Stringer = (*Diagnostic)(nil)
+
+func (d Diagnostic) String() string {
+	result := fmt.Sprintf("(%s) %s", d.Severity.String(), d.Summary)
+	if d.Detail == "" {
+		return result
+	}
+
+	return result + `
+` + d.Detail
```

Comment:
Nit: Is that for a new line? I thought a `\n` might be nicer ;)

Reply:
ya, I was looking for it to not be OS dependent but if there's a better golang trick I'm all ears

Reply:
Go's standard `fmt.Sprintln` should do the right thing. I haven't been on Windows for a while, but I vaguely remember that some IDEs would store files with Unix line endings anyway, so I guess it may be an issue either way...

Reply:
much better, thanks